src/ORM/Behavior.php

```diff
@@ -188,6 +188,8 @@ public function initialize(array $config): void
      */
     public function getTable(): Table
     {
+        deprecationWarning('Behavior::getTable() is deprecated. Use table() instead.');
```

Comment:
Are we going to throw a warning the same point release it was deprecated? I am not sure what the policy is or if there is one.

Reply:
Throwing them a bit too soon can be annoying for plugins and stuff.
As they always need to then pin to the highest version of minors, leaving a lot of people very annoyed by this.
The same issue I mentioned back when we started to do hard instead of soft deprecations (or rather opt out vs opt in).

Reply:
What benefit would delaying warnings have? For anyone keeping up to date with the minors will eventually see the warnings anyway.

Reply:
Plugins wouldnt be forced to directly upgrade, and minimize constraint range.
So a wider constraint range makes it easier to support a wider range of users and their different plugin usage.
All they need would be to silence. But if plugins are force-constraint to last minor all the time, this puts a lot of pressure on all involved and makes it impossible to keep a lot of different plugins in the app - since not all plugins are well maintained.

Reply:
I don't see how these deprecation warnings force constraint to last minor when one can just silence the deprecations by modifying the error level.

I wouldn't mind waiting till the last minor release to add deprecation warnings but we don't follow fixed release cycles and don't exactly know which minor will be the last one. We started adding them in 3.6 thinking it might be the last but we are up to 3.9 now. Also waiting till next minor to add the warning doesn't work in practice as evident by this PR, we always forget to add the warnings later.

Reply:
Thats because you only have a very small amount of plugins I do :)
I can tell you, this is super annoying.

The tests are failing, so you remove the deprecated part in favor of the new code.
But now you need to constraint, as the code would not work anymore with the pervious minor
So this is the force on last minor for all plugins that have anything related to this kind of deprecation errors thrown.

Reply:
This all wouldnt be an issue if we went to what I advocating very loudly for years:
Opt-in instead of Opt-out notifications :)

Reply:
I agree. With PHP, I'd much prefer warnings in doc blocks.

Reply:
@dereuromark Building another mechanism on top of PHP errors seems silly when we already have access to an opt-in/out system for error warnings. Should we be setting the default error level in plugins and applications to exclude deprecations?

If people think the runtime warnings are annoying and not helpful, we can skip this process entirely and do only documentation warnings. I don't think documentation only warnings will help users though. You can't use documentation tags to verify that your application has no remaining deprecations. Runtime warnings lets you at least be confident that the code you have tests for, or code you manually test no longer has deprecations.